tiny_cnn/layers/max_pooling_layer.h

```diff
@@ -45,9 +45,11 @@ class max_pooling_layer : public feedforward_layer<Activation> {
         in_(in_width, in_height, in_channels),
         out_(in_width / pooling_size, in_height / pooling_size, in_channels)
     {
-        if ((in_width % pooling_size) || (in_height % pooling_size))
+        if ((in_width % pooling_size) || (in_height % pooling_size)) {
             pooling_size_mismatch(in_width, in_height, pooling_size);
+        }
 
+        set_worker_count(CNN_TASK_SIZE);
```

Comment:
@edgarriba
Calling set_worker_count in the constructor will cause segv in this branch. Removing this line perhaps improve the problem